quantumdb-postgresql/src/main/java/io/quantumdb/core/planner/SyncFunction.java

```diff
@@ -112,13 +112,13 @@ public void setColumnsToMigrate(Set<String> columnsToMigrate) {
 		this.insertExpressions = ImmutableMap.copyOf(expressions);
 		this.updateExpressions = ImmutableMap.copyOf(insertExpressions);
 
-		this.updateIdentitiesForInserts = ImmutableMap.copyOf(targetTable.getIdentityColumns().stream()
+		this.updateIdentitiesForInserts = ImmutableMap.copyOf((Map<? extends String, ? extends String>)targetTable.getIdentityColumns().stream()
```

Comment:
Just to verify could you let me know which JDK you're using? We've seen this before at Magnet.me, and this only occurred on specific older versions of the Java SDK. We guessed the compiler loses type information at some point, and that makes IntelliJ complain requiring an explicit cast where it's not really needed.

Reply:
My IntelliJ project uses openjdk-15 Java version "15.0.2"

Reply:
Interesting. Let me get back to you on that one.

Reply:
Ok, I can't reproduce it with that specific version of openjdk. Most likely it's due to a config/setting in IntelliJ. Probably best to debug that over a video call instead of Github. Until then, let's leave this cast in.